StellarApp/Protocol/Planet.swift

```diff
@@ -0,0 +1,220 @@
+                
+            case .neptune:
+                let orbitNode = SCNNode.getRing(radius: 15.98, vector: (x: 0, y: 0, z: 0), color: .yellow)
+                let planetNode = SCNNode.getNode(name: "neptune", image: UIImage(named: "art.scnassets/neptune.jpg")!, radius: 0.5, vector: (x: 15.98, y: 0, z: 0), moveSpeed: 0.4, rotation: 0.8)
+                orbitNode.addChildNode(planetNode)
+                planetNodes.append(orbitNode)
+                
+            case .pluto:
+                let orbitNode = SCNNode.getRing(radius: 18.65, vector: (x: 0, y: 0, z: 0), color: .yellow)
+                let planetNode = SCNNode.getNode(name: "pluto", image: UIImage(named: "art.scnassets/pluto.jpg")!, radius: 0.19, vector: (x: 18.65, y: 0, z: 0), moveSpeed: 0.4, rotation: 0.8)
+                orbitNode.addChildNode(planetNode)     
+                planetNodes.append(orbitNode)
+
+            }
+        }
+        return planetNodes
+    }
+}
+
+extension SCNNode {
+    static func deepScaleNode(node: SCNNode, scaleAction: SCNAction) {
+        node.runAction(scaleAction)
+        for child in node.childNodes {
+            deepScaleNode(node: child, scaleAction: scaleAction)
+        }
+    }
+    
+    static func deepScaleNode(node: SCNNode, scale: CGFloat) {
+        let scaleX = scale * CGFloat(node.scale.x)
+        let scaleY = scale * CGFloat(node.scale.y)
+        let scaleZ = scale * CGFloat(node.scale.z)
+        
+        let newScale = SCNVector3Make(Float(scaleX), Float(scaleY), Float(scaleZ))
+        node.scale = newScale
+        for child in node.childNodes {
+            deepScaleNode(node: child, scale: scale)
+        }
+    }
+    
+    
+    
+    static func getNode(name: String, image: UIImage,
+                        radius: CGFloat,
+                        vector: (x: Float, y: Float, z: Float), moveSpeed: CGFloat, rotation: CGFloat) -> SCNNode{
+        
+        let node = SCNNode()
+        let nodeSize = SCNSphere(radius: radius)
+        let material = SCNMaterial()
+        material.diffuse.contents = image
+        node.name = name
+        nodeSize.materials = [material]
+        node.position = SCNVector3(x: vector.x, y: vector.y, z: vector.z)
+        node.geometry = nodeSize
+        node.runAction(SCNAction.repeatForever(SCNAction.rotateBy(x: 0, y: rotation, z: 0, duration: 1)))
+        
+        return node
+    }
+    
+    static func getRing(radius: CGFloat,
+                        vector: (x: Float, y: Float, z: Float),
+                        color: UIColor) -> SCNNode {
+        
+        var ringGeo = SCNTorus()
+        let ringNode = SCNNode()
+        
+        ringGeo = SCNTorus(ringRadius: radius, pipeRadius: 0.01)
+        ringNode.geometry = ringGeo
+        ringNode.position = SCNVector3(x: vector.x, y: vector.y, z: vector.z)
+        ringGeo.firstMaterial?.diffuse.contents = color
+        
+        return ringNode
+    }
+    
+    static func getSaturnRing(radius: CGFloat,
```

Comment:
this is the exact same function as the one above it. Get rid of this one and use the more generic one

Reply:
ok sounds good. Then maybe just remove the `Saturn` part. `getRing`